File: system-test/typescript/src/testkit/dsl/port/then/steps/then-order.ts
import type { ThenStep } from './then-step.js';

export interface ThenOrder extends ThenStep<ThenOrder> {
  hasSku(expectedSku: string): ThenOrder;
  hasQuantity(expectedQuantity: number): ThenOrder;
  hasUnitPrice(expectedUnitPrice: number): ThenOrder;
  hasBasePrice(expectedBasePrice: string | number): ThenOrder;
  hasSubtotalPrice(expectedSubtotalPrice: string | number): ThenOrder;
  hasTotalPrice(expectedTotalPrice: string | number): ThenOrder;
  hasStatus(expectedStatus: string): ThenOrder;
  hasTotalPriceGreaterThanZero(): ThenOrder;
  hasOrderNumberPrefix(expectedPrefix: string): ThenOrder;
  hasDiscountRate(expectedDiscountRate: number): ThenOrder;
  hasDiscountAmount(expectedDiscountAmount: string | number): ThenOrder;
  hasAppliedCouponCode(expectedCouponCode: string): ThenOrder;
  hasAppliedCoupon(expectedCouponCode?: string): ThenOrder;
  hasTaxRate(expectedTaxRate: string | number): ThenOrder;
  hasTaxAmount(expectedTaxAmount: string): ThenOrder;
}

File: system/multitier/backend-clean-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/then/steps/ThenOrder.java
package com.mycompany.myshop.backend.testkit.dsl.port.then.steps;

import com.mycompany.myshop.backend.domain.values.OrderStatus;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.base.ThenStep;

public interface ThenOrder extends ThenStep<ThenOrder> {
    ThenOrder hasSku(String expectedSku);

    ThenOrder hasQuantity(int expectedQuantity);

    ThenOrder hasUnitPrice(String expectedUnitPrice);

    ThenOrder hasUnitPrice(double expectedUnitPrice);

    ThenOrder hasBasePrice(String expectedBasePrice);

    ThenOrder hasBasePrice(double expectedBasePrice);

    ThenOrder hasDiscountRate(String expectedDiscountRate);

    ThenOrder hasDiscountRate(double expectedDiscountRate);

    ThenOrder hasTaxRate(String expectedTaxRate);

    ThenOrder hasTaxRate(double expectedTaxRate);

    ThenOrder hasOrderNumberPrefix(String expectedPrefix);

    ThenOrder hasDiscountAmount(String expectedDiscountAmount);

    ThenOrder hasDiscountAmount(double expectedDiscountAmount);

    ThenOrder hasSubtotalPrice(String expectedSubtotalPrice);

    ThenOrder hasSubtotalPrice(double expectedSubtotalPrice);

    ThenOrder hasTaxAmount(String expectedTaxAmount);

    ThenOrder hasTaxAmount(double expectedTaxAmount);

    ThenOrder hasTotalPrice(String expectedTotalPrice);

    ThenOrder hasTotalPrice(double expectedTotalPrice);

    ThenOrder hasStatus(OrderStatus expectedStatus);

    ThenOrder hasAppliedCoupon(String expectedCouponCode);

    ThenOrder hasAppliedCoupon();

    ThenOrder hasNoAppliedCoupon();
}
